File: 1833-find-the-highest-altitude/find-the-highest-altitude.ts
function largestAltitude(gain: number[]): number {
    const n = gain.length;
    let prefix = new Array(n+1).fill(0);
    for (let i = 1;i<prefix.length;i++){
        prefix[i] = gain[i-1]+prefix[i-1]
    }
    return Math.max(...prefix);
};

File: 1833-find-the-highest-altitude/find-the-highest-altitude.py
class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        n = len(gain)
        amps = [0]*(n+1)
        for i in range(1,len(amps)):
            amps[i] = gain[i-1]+amps[i-1]
        return max(amps)